File: src/main/java/com/thegame/ui/controller/ResourceRestController.java
package com.thegame.ui.controller;

import com.thegame.business.dto.ResourceUpdateRequestDTO;
import com.thegame.business.model.Resource;
import com.thegame.business.service.ResourceService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/resource")
public class ResourceRestController {

    private final ResourceService resourceService;

    public ResourceRestController(ResourceService resourceService) {
        this.resourceService = resourceService;
    }

    @GetMapping("/getAll")
    public List<Resource> getAllResources() {
        return resourceService.getAllResources();
    }

    @PutMapping("/update")
    public void updateResources(@RequestBody ResourceUpdateRequestDTO request) {
        resourceService.updateResourcesByVillageId(request);
    }
}

File: src/main/java/com/thegame/ui/controller/ResourceRestController.kt
package com.thegame.ui.controller

import com.thegame.business.model.Resource
import com.thegame.business.service.ResourceService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/resource")
class ResourceRestController(private val resourceService: ResourceService) {

    @GetMapping("/getAll")
    fun getAllResources(): List<Resource> {
        return resourceService.getAllResources()
    }

    @PutMapping("/update")
    fun updateResources() {
        return resourceService.updateResourcesByVillageId()
    }
}